### PR TITLE
Refactor HostgameWidget

### PR DESCRIPTION
**Pull request by yorick-ne**
Moved lots of similar code from GamesWidget and CoopWidget over to HostgameWidget
This should fix hosting for coop games
Also fixes a bug that prevented joining coop games ( untested)

### CODE REVIEW

File: src/games/hostgamewidget.py [hunk 1]
```diff
@@ -86,19 +101,56 @@ def hosting(self):
...
         if self.passCheck.isChecked():
-            password = self.passEdit.text()
+            self.password = self.passEdit.text()
         else:
-            password = None
+            self.password = None
+        self.save_last_hosted_settings()
+
+
+        self.save_last_hosted_settings()
```
Comment (Sheeo) [inline src/games/hostgamewidget.py]: This seems redundant?

File: src/games/hostgamewidget.py [hunk 2]
```diff
@@ -86,19 +101,56 @@ def hosting(self):
...
 
     def mapChanged(self, index):
-        self.parent.gamemap = self.mapList.itemData(index)
+        self.mapname = self.mapList.itemData(index)
 
-        self.message['mapname'] = self.parent.gamemap
+        self.message['mapname'] = self.mapname
         self.game.update(self.message, self.parent.client)
+
+    def save_last_hosted_settings(self):
```
Comment (Sheeo) [inline src/games/hostgamewidget.py:144]: Looks like this method still exists on `GamesWidget`, is that intended?

Would like to see these migrated to use the persisted properties sometime in the future, out of scope of this PR though :)

**Comment by Sheeo** [comment]
Looks good, would like to test that it fixes coop hosting on the test server.

File: src/games/hostgamewidget.py (hunk 2 above)
Comment (yorick-ne) [inline src/games/hostgamewidget.py:144]: I think i just forgot to remove that method + 'load_last_hosted_settings' on the Gameswidget class. They didnt appear to get called anywhere.

**Comment by speed2CZ** [comment]
When hosting game I cant change title of the lobby, no matter what I type there it uses the one I had before this fix.

**Comment by HardlySoftly** [comment]
I also had the same issue as @speed2CZ

**Comment by yorick-ne** [comment]
@speed2CZ @HardlySoftly fixed now, thanks for the report

**Comment by yorick-ne** [comment]
@arkitect-dev why has this been closed ?

**Comment by arkitect-dev** [comment]
sorry, waffle closed automatically when i dragged it into another column

**Comment by Sheeo** [comment]
👍

**Pull request merged**